blockchain/indexers/utreexobackend.go

```diff
@@ -157,111 +135,180 @@ func (idx *FlatUtreexoProofIndex) FlushUtreexoState() error {
 	if err != nil {
 		return err
 	}
-	err = idx.utreexoState.state.WriteForestToDisk(forestFile, true, false)
-	if err != nil {
-		return err
-	}
-
-	miscFilePath := filepath.Join(basePath, defaultUtreexoMiscFileName)
-	// write other misc forest data
-	miscForestFile, err := os.OpenFile(
-		miscFilePath, os.O_RDWR|os.O_CREATE, 0666)
-	if err != nil {
-		return err
-	}
-	err = idx.utreexoState.state.WriteMiscData(miscForestFile)
+	_, err = idx.utreexoState.state.WriteTo(forestFile)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-// restoreUtreexoState restores forest fields based off the existing utreexo state
-// on disk.
-func restoreUtreexoState(cfg *UtreexoConfig, basePath string) (
-	*accumulator.Forest, error) {
+func serializeUtreexoRoots(p *utreexo.Pollard) ([]byte, error) {
+	w := bytes.NewBuffer(make([]byte, 0, 8+(len(p.Roots)*chainhash.HashSize)))
 
-	var forest *accumulator.Forest
+	var buf [8]byte
+	byteOrder.PutUint64(buf[:], p.NumLeaves)
 
-	switch cfg.Type {
-	case accumulator.CowForest:
-		cowPath := filepath.Join(basePath, defaultUtreexoCowDirName)
-		miscForestFilePath := filepath.Join(basePath, defaultUtreexoMiscFileName)
+	_, err := w.Write(buf[:])
+	if err != nil {
+		return nil, err
+	}
 
-		// Where the misc forest data exists
-		miscForestFile, err := os.OpenFile(
-			miscForestFilePath, os.O_RDONLY, 0400)
+	for _, root := range p.GetRoots() {
+		_, err = w.Write(root[:])
 		if err != nil {
 			return nil, err
 		}
-		forest, err = accumulator.RestoreForest(
-			miscForestFile, nil, false, false, cowPath, defaultCowMaxCache)
+	}
+
+	return w.Bytes(), nil
+}
+
+func deserializeUtreexoRoots(serialized []byte) (utreexo.Stump, error) {
```

Comment:
Should get rid of this since it's a duplicate of the `DeserializeUtreexoRoots` in `blockchain`.